internal/services/mssqlmanagedinstance/mssql_managed_instance_resource_test.go

```diff
@@ -1359,6 +1394,11 @@ resource "azurerm_mssql_managed_instance" "test" {
     environment = "staging"
     database    = "test"
   }
+
+  # Changing azure_active_directory_administrator is ignored because API returns the values of azure_active_directory_administrator even if it is not specified in the config when microsoft entra authentication only is enabled
```

Comment:
`ignore_changes` is added here because test case `TestAccMsSqlManagedInstanceActiveDirectoryAdministrator_basic` also calls this identity method [here](https://github.com/hashicorp/terraform-provider-azurerm/blob/ce4cc5715476162c8da281f6b66cd91500739b07/internal/services/mssqlmanagedinstance/mssql_managed_instance_active_directory_administrator_resource_test.go#L86). The test case failed after it applied this PR, as detailed below. I assume this is the correct behavior since this is equivalent to enable AAD Auth-only through another resource `azurerm_mssql_managed_instance_active_directory_administrator` after the `azurerm_mssql_managed_instance` is successfully created (without enabling AAD Auth-only). Then it is normal to get this diff when refreshing again . 

![image](https://github.com/hashicorp/terraform-provider-azurerm/assets/39109137/ad02250f-b74e-4e00-9e42-bfbc970d496f)

However, thank you very much for asking this question. It is really inappropriate to add `ignore_changes` in `identity`, so I deleted `ignore_changes` in `identity` and modified test case `TestAccMsSqlManagedInstanceActiveDirectoryAdministrator_basic` to avoid confusion by not sharing the `identity` function between `azurerm_mssql_managed_instance_active_directory_administrator` and `azurerm_mssql_managed_instance`. Any feedback please feel free to let me know. Thanks again.